### DISCUSSION TITLE
How to set message expiry time interval -> https://www.emqx.com/en/blog/mqtt-message-expiry-interval in emqx.conf

### DISCUSSION THREAD

**Discussion by haripriyapendem** [emqx/emqx#13223] [Q&A] [answered]
How to set message expiry time interval -> https://www.emqx.com/en/blog/mqtt-message-expiry-interval in emqx.conf.

MQTT was delaying messages. For example, publisher sends a message to topic1 at 9:10. mqtt server receives this message around 9:20. subscriber receives this message on topic1 at 9:21. Here message is delayed by 10 minutes. How to reduce the delays. How to scale up and rebalance load with out affecting current connections?

**Comment by qzhuyan** [comment]
`expiry time interval`  is set by the client not by EMQX.

How big is the payload? is the client under poor network? Maybe do some tcpdump to see if there are many retransmissions.

**Comment by zmstone** [comment]
client only specifies the message expiry interval, but not the message expiry timestamp (deadline).
EMQX records the timestamp AFTER a full PUBLISH message is received.
Meaning the time spent for the client to transmit all the bytes are not included in the expiry.

**Comment by zmstone** [comment, marked as answer]
For messages lack of a `Message-Expiry-Interval`, EMQX assigns it from the config which has a default value `infinity` (never expire).
Config ref here: https://docs.emqx.com/en/enterprise/v5.7.0/hocon/#V-mqtt-S-mqtt-message_expiry_interval

in emqx.conf:
`mqtt.message_expiry_interval = infinity`
or
`mqtt.message_expiry_interval = 600s`
